test/run-spec-wasm2c.py

```diff
@@ -165,7 +187,73 @@ def _CacheModulePrefixes(self):
                 else:
                     name_idx = idx - 1
 
+                if name_idx in self.idx_to_module_name.keys():
+                    self.module_prefix_map[self.idx_to_module_name[name_idx]] = name
                 self.module_prefix_map[name_idx] = name
+                c_filename = command['as'] + '.c'
+                self.idx_to_c_name[name_idx] = c_filename
+
+    def _WriteModuleInstanceImports(self, command, uninstantiable):
+        c_filename = os.path.join(self.out_dir, self.idx_to_c_name[self.module_idx - 1])
+        header_filename = utils.ChangeExt(c_filename, '.h')
+        with open(header_filename, encoding='utf-8') as f:
+            headerfile = f.readlines()
+            imported_modules = set()
+            for i in range(0, len(headerfile)):
+                line = headerfile[i]
+                if "import: " in line:
+                    line_split = line.split()
+                    import_module_name = MangleName(line_split[2][1:-1])
+                    imported_modules.add(import_module_name)
+
+            if (uninstantiable):
+                self.out_file.write("ASSERT_TRAP(")
+
+            if (len(imported_modules) > 0):
+                self.out_file.write("%s_init(&%s_module_instance" % (self.GetModulePrefix(), self.GetModulePrefix()))
+                for imported_module in sorted(imported_modules):
+                    self.out_file.write(", &%s_module_instance" % imported_module)
+                self.out_file.write(")")
+            else:
+                self.out_file.write("%s_init(&%s_module_instance)" % (self.GetModulePrefix(), self.GetModulePrefix()))
+
+            if (uninstantiable):
+                self.out_file.write(")")
+            self.out_file.write(";\n")
+
+    def _WriteModuleImports(self):
+        for idx, c_filename in self.idx_to_c_name.items():
+            header_filename = utils.ChangeExt(os.path.join(self.out_dir, c_filename), '.h')
+
+            with open(header_filename, encoding='utf-8') as f:
+                headerfile = f.readlines()
+                imported_functions = {}
+                for i in range(0, len(headerfile)):
```

Comment:
same here

Reply:
This one seems trickier because we want to look ahead one line to detect a function import. Added a comment to clarify (https://github.com/fixpointOS/wabt/blob/module_instancing/test/run-spec-wasm2c.py#L223-L224).